tests/listeners/class_listener.py

```diff
@@ -41,6 +41,49 @@ def before_stopping(self, component):
             stopped_component = component
             self.state.append(DagRunState.SUCCESS)
 
+        @hookimpl
+        def on_task_instance_running(self, previous_state, task_instance):
+            self.state.append(TaskInstanceState.RUNNING)
+
+        @hookimpl
+        def on_task_instance_success(self, previous_state, task_instance):
+            self.state.append(TaskInstanceState.SUCCESS)
+
+        @hookimpl
+        def on_task_instance_failed(self, previous_state, task_instance, error: None | str | BaseException):
+            self.state.append(TaskInstanceState.FAILED)
+
+        @hookimpl
+        def on_dag_run_running(self, dag_run, msg: str):
+            self.state.append(DagRunState.RUNNING)
+
+        @hookimpl
+        def on_dag_run_success(self, dag_run, msg: str):
+            self.state.append(DagRunState.SUCCESS)
+
+        @hookimpl
+        def on_dag_run_failed(self, dag_run, msg: str):
+            self.state.append(DagRunState.FAILED)
+
+elif AIRFLOW_V_2_10_PLUS:
```

Comment:
Will we remove this code path for airflow 3 release ?

Reply:
Or do we need to keep that for provider compatibility ?

Reply:
It is here for provider compatibility. I think it can be moved to particular provider tho: would require more unrelated test changes and I'd rather do them in a separate PR.

Reply:
Sound good to me, thanks.